Week3/js-exercises/ex5-LotteryMachine.js

```diff
@@ -28,8 +28,22 @@ Don't you just love the thrill of the lottery? What if I told you we can make ou
 
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
-  // make array
-  // start at beginning of array and check if you should call threeCallback or fiveCallback or go on to next
+  for( let i = startIndex; i <= stopIndex; i++){
+    numbers.push(i);
+  }
+
+  console.log(numbers);
+
+  for(let i = 0; i < numbers.length; i++){
+    if ( numbers[i] % 3 === 0 && numbers[i] % 5 !== 0 ) {
+      threeCallback();
+    } else if ( numbers[i] % 5 === 0 && numbers[i] % 3 !== 0) {
+      fiveCallback();
+    } else if ( numbers[i] % 3 === 0 && numbers[i] % 5 === 0 ) {
+      threeCallback();
+      fiveCallback();
```

Comment:
This construction of if/elseif definitely works to catch the there situations (only divisible by three, only five, and both) You could also consider this code:
```suggestion
    if ( numbers[i] % 3 === 0 ) {
      threeCallback();
    } 
    if ( numbers[i] % 5 === 0) {
      fiveCallback();
    } 
```
Two consecutive `if` statements are independant and will both fire if both conditions are met. Choose whatever you like best 🙂

Reply:
Oh yes you are right. I have updated the conditions :)